### PR TITLE
MOD-7323: Support amazonlinux 2023

### PR DESCRIPTION
**Pull request by nafraf**
Add support for amazonlinux:2023 for amd64 and arm64.
`build-redisjson` now installs rust without using `$READIES/bin/getrust` because it does not support recent OS versions.

**Which issues this PR fixes**
1. MOD-7323

**Main objects this PR modified**
1. Github workflows

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: sbin/build-redisjson [hunk 1]
```diff
@@ -54,22 +54,27 @@ $OP cd RedisJSON
...
 	. venv/bin/activate
 fi
-# Install only rust, because the rest of the dependencies are already installed
-runn $READIES/bin/getrust
-$OP source /etc/profile.d/rust.sh
 
-runn BINROOT=$TARGET_DIR make SAN=$SAN
+# Install only rust, because the rest of the dependencies are already installed
+if [[ $NOP != 1 ]]; then
+	runn "curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y"
```
Comment (GuyAv46) [inline sbin/build-redisjson:66]: Great. Can we also not use `runn` (a readies command) to complete the removal of readies here?

File: sbin/build-redisjson [hunk 2]
```diff
@@ -54,22 +54,27 @@ $OP cd RedisJSON
...
+	source $HOME/.cargo/env
+	rustup update 
+	rustup update nightly
+	# for RedisJSON build with addess santizer
+	rustup component add rust-src --toolchain nightly
+	rustup show 
+
+	runn BINROOT=$TARGET_DIR make SAN=$SAN
```
Comment (GuyAv46) [inline sbin/build-redisjson:74]: Same here

File: .github/workflows/task-get-linux-configurations.yml [hunk 3]
```diff
@@ -107,6 +109,16 @@ jobs:
               'OS': 'amazonlinux:2',
               'pre-deps': "yum install -y tar gzip"})
 
+          # amazonlinux:2023 needs pre-checkout dependencies
+          if x86_platforms.count('amazonlinux:2023') > 0:
```
Comment (raz-mon) [inline .github/workflows/task-get-linux-configurations.yml:113]: Why `count` and not `in`?

Comment (raz-mon) [reply .github/workflows/task-get-linux-configurations.yml:113]:
```suggestion
          if 'amazonlinux:2023' in x86_platforms:
```

**Review by raz-mon — COMMENTED** [review]
Nice 🚀 
Please attach a link to the build & test flow itself (since it is not in the PR flow)

File: .github/workflows/task-get-linux-configurations.yml (hunk 3 above)
Comment (GuyAv46) [reply .github/workflows/task-get-linux-configurations.yml:113]: I think it was copilot suggestion… let’s fix it on all the cases so they are aligned

File: sbin/build-redisjson (hunk 1 above)
Comment (nafraf) [reply sbin/build-redisjson]: OK. Done. Thanks.

File: sbin/build-redisjson (hunk 2 above)
Comment (nafraf) [reply sbin/build-redisjson]: Done.

File: .github/workflows/task-get-linux-configurations.yml (hunk 3 above)
Comment (nafraf) [reply .github/workflows/task-get-linux-configurations.yml]: Done.

**Comment by nafraf** [comment]
> Nice 🚀 Please attach a link to the build & test flow itself (since it is not in the PR flow)

For all OS/archs except bionic (flaky -  it failed installing packages):
https://github.com/RediSearch/RediSearch/actions/runs/9894329721

For bionic:
https://github.com/RediSearch/RediSearch/actions/runs/9897185523

**Review by GuyAv46 — APPROVED** [review]

**Pull request merged**